src/crawler.js

```diff
@@ -71,22 +75,43 @@ function normalizeIpp(ip, port) {
 
 function Crawler(maxRequests, logger) {
   EventEmitter.call(this);
-  this.maxRequests = maxRequests ? maxRequests : 30;
+  
+ 
+  // maxRequests checks
+  maxRequests = maxRequests ? maxRequests : 30;
+  if (!maxRequests || maxRequests < 1) {
+    throw new Error("Invalid max requests");
+  }
+  check.assert.number(maxRequests, "Invalid max requests");
+  
+  // logger checks
+  logger = logger ? logger : console
+  if (!logger.log || !logger.error) {
+    throw new Error("Invalid logger");
```

Comment:
TypeError is the more appropriate type

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError